crates/api/src/op_store.rs

```diff
@@ -0,0 +1,86 @@
+//! Kitsune2 op store types.
+
+use crate::{K2Result, OpId, Timestamp};
+use futures::future::BoxFuture;
+use std::cmp::Ordering;
+use std::sync::Arc;
+
+/// An op with metadata.
+///
+/// This is the basic unit of data in the kitsune2 system.
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct MetaOp {
+    /// The id of the op.
+    pub op_id: OpId,
+
+    /// The creation timestamp of the op.
+    ///
+    /// This must be the same for everyone who sees this op.
+    ///
+    /// The host must reject the op if the timestamp does not agree with any timestamps inside the
+    /// op data.
+    pub timestamp: Timestamp,
```

Comment:
Want to expose this outside the op data because we want a concept of "when an op was created" but it's duplicated inside the op data. Suggestions? 

One option would be to make this field test only and we don't know when and op is received what it's timestamp is. Then have another op type for gossip to use where the timestamp is provided.

Reply:
Since kitsune can't parse the op content anyways... we would need to either:
- always hand raw ops received directly to the host, and get back out the MetaOp as part of querying the host with the timestamp included
- or, have the host provide some kind of parsing function when initializing kitsune that could extract the timestamp

I feel like we want the timestamp and any other data we need to be explicitly available to kitsune because that will simplify our implementation, but you should tell me what you think here.

Reply:
I like the first version. We shouldn't be doing anything with ops that haven't been persisted on the host. So whether we wait for persistence or integration, I think we need to hand ops to the host and have the host tell us when it's "processed" them. Whatever that means to a given host.

Reply:
Exactly that, though based on this we'd need to make some changes to the types used in that

Reply:
https://github.com/holochain/kitsune2/issues/31